test-app/backend/src/shift/shift-api.ts

```diff
@@ -0,0 +1,45 @@
+import app from '../server';
+import isUserAdmin from '../middleware';
+import Database from '../models/database/database';
+import shift from './shift-interface';
+
+app.post('/api/new-shift', isUserAdmin, async (req, res) => {
+  console.log('POST: Creating Calendar Event...');
+
+  const { start, end, maxVolunteers, title, description, eventAdmin } = req.body as shift;
```

Comment:
The start and end dates here are actually going to be given as numbers, not `Date`s. We just need to convert those numbers to dates after confirming the type. Perhaps a function called like `shiftPreProcessor` that takes a `Partial<Shift>` and goes through and converts the numbers to actual dates and returns the edited shift.